smart-contracts/wasm-chain-integration/src/utils.rs

```diff
@@ -273,34 +279,30 @@ impl<'a, R: RngCore, BackingStore: trie::BackingStoreLoad> machine::Host<Artifac
                 self.slot_time = Some(slot_time);
             }
             "get_slot_time" => {
-                let slot_time = self.slot_time.context(unset_err("slot_time"))?;
+                let slot_time = self.slot_time.ok_or(CallErr::Unset("slot_time"))?;
                 stack.push_value(slot_time);
             }
             "set_receive_self_address" => {
                 let addr_ptr = unsafe { stack.pop_u32() };
 
                 let mut cursor = Cursor::new(memory);
-                cursor.seek(SeekFrom::Start(addr_ptr)).map_err(|_| anyhow!(seek_err))?;
+                cursor.seek(SeekFrom::Start(addr_ptr)).map_err(|_| CallErr::Seek)?;
 
                 self.address = Some(ContractAddress::deserial(&mut cursor)?);
             }
             "get_receive_self_address" => {
-                let addr_ptr = unsafe { stack.pop_u32() };
-                let mut cursor = Cursor::new(memory);
-
-                cursor.seek(SeekFrom::Start(addr_ptr)).map_err(|_| anyhow!(seek_err))?;
-
+                let addr_ptr = unsafe { stack.pop_u32() } as usize;
                 self.address
-                    .context(unset_err("slot_time"))?
-                    .serial(&mut cursor)
+                    .ok_or(CallErr::Unset("address"))?
+                    .serial(&mut &mut memory[addr_ptr..])
```

Comment:
```suggestion
                    .serial(&mut memory[addr_ptr..])
```

Reply:
Also my thought, but `&mut memory[addr_ptr..]` is a `[u8]` and `&mut &mut memory[addr_ptr..]` is the desired `&mut [u8]`. No idea why though, but I think it's something funky with slices :shrug:

![image](https://github.com/user-attachments/assets/bf4cdbd8-dd90-44d5-9164-2ebe85070e14)

Reply:
huh... okay then :+1: